src/templates/pageDeprecatedFiltersDrawer.jsx

```diff
@@ -170,14 +172,83 @@ class PageFiltersDrawer extends React.Component {
-        this._onMultiSelectChange = this._onMultiSelectChange.bind(this);
-        this._onNestedTogglesCloseWingClick = this._onNestedTogglesCloseWingClick.bind(this);
-        this._onNestedTogglesLabelClick = this._onNestedTogglesLabelClick.bind(this);
-        this._onNestedTogglesWingOptionLabelClick = this._onNestedTogglesWingOptionLabelClick.bind(this);
-        this._onNestedTogglesValueLabelClick = this._onNestedTogglesValueLabelClick.bind(this);
+        this.onApplyClick = this.onApplyClick.bind(this);
+        this.onClearClick = this.onClearClick.bind(this);
+        this.onMultiSelectLabelClearClick = this.onMultiSelectLabelClearClick.bind(this);
+        this.onMultiSelectChange = this.onMultiSelectChange.bind(this);
+        this.onNestedTogglesCloseWingClick = this.onNestedTogglesCloseWingClick.bind(this);
+        this.onNestedTogglesLabelClick = this.onNestedTogglesLabelClick.bind(this);
+        this.onNestedTogglesWingOptionLabelClick = this.onNestedTogglesWingOptionLabelClick.bind(this);
+        this.onNestedTogglesValueLabelClick = this.onNestedTogglesValueLabelClick.bind(this);
+    }
+
+    onApplyClick() {
+        const { onApply } = this.props;
+
+        onApply();
+    }
+
+    onClearClick() {
+        const { onClear } = this.props;
+
+        onClear();
+    }
+
+    onMultiSelectLabelClearClick(onItemChange, value, selectedOption) {
+        const filteredOptions = _.differenceBy(value, [ selectedOption ], 'value');
+
+        onItemChange(filteredOptions);
+    }
+
+    onMultiSelectChange(onItemChange, value, selectedOption) {
+        const filteredOptions = _.union(value, [ selectedOption ]);
+
+        onItemChange(filteredOptions);
+    }
+
+    onNestedTogglesCloseWingClick() {
+        const { nestedTogglesData } = this.state;
+
+        this.setState({
+            nestedTogglesData: {},
+        }, () => {
+            nestedTogglesData.onChange(nestedTogglesData.value);
+        });
+    }
+
+    onNestedTogglesLabelClick(nestedTogglesData) {
+        this.setState({
+            nestedTogglesData: _.cloneDeep(nestedTogglesData),
+        });
+    }
+
+    onNestedTogglesValueLabelClick(nestedTogglesData, selectedOption) {
+        const selectedOptions = _.filter(nestedTogglesData.value, d => d.value !== selectedOption.value);
+
+        nestedTogglesData.onChange(selectedOptions);
+    }
+
+    onNestedTogglesWingOptionLabelClick(selectedOption) {
+        const { nestedTogglesData } = this.state;
+        let selectedOptions;
+
+        if (_.some(nestedTogglesData.value, selectedOption)) { // Subtract
+            selectedOptions = _.filter(nestedTogglesData.value, d => d.value !== selectedOption.value);
+        } else { // Add
+            selectedOptions = _.sortBy([ ...nestedTogglesData.value, selectedOption ], [ 'value' ]);
+        }
+
+        const newNestedTogglesData = _.mapValues(nestedTogglesData, (d, key) => {
+            if (key === 'value') {
+                d = selectedOptions;
```

Comment:
We're intentionally assigning to (mutating) an iteration variable here?  That's unusual.  But it is the correct logic and is being done intentionally?

Reply:
It was intentional, not saying it was the best though.